### PR TITLE
Secondary Invigilator allotment due to lack of Primary invigilators

### PR DESCRIPTION
**Pull request by amartya2969**
Both the secondary invigilators to be allotted  in the room in case of shortage of primary invigilators

### CODE REVIEW

File: MainFrame.py [hunk 1]
```diff
@@ -171,71 +171,71 @@ def __init__( self, parent ):
 
 		gbSizer211.Add( self.m_staticText111, wx.GBPosition( 0, 0 ), wx.GBSpan( 1, 1 ), wx.ALIGN_CENTER_VERTICAL|wx.ALL, 5 )
 
-		self.invig_room_csv_picker = wx.FilePickerCtrl( self.m_panel5, wx.ID_ANY, wx.EmptyString, u"Select a file", u"*.csv", wx.DefaultPosition, wx.DefaultSize, wx.FLP_DEFAULT_STYLE )
```
Comment (amartya2969) [inline MainFrame.py]: Input location needed to chosen

File: algorithms/Invigilation/Course.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def add(self, code, name, ic, faculty_list):
 
         if code is None:
             return
-        
+
```
Comment (Kailash0311) [inline algorithms/Invigilation/Course.py]: Changes in this file need to be reverted.

File: algorithms/Invigilation/Invigilator.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+import datetime
```
Comment (Kailash0311) [inline algorithms/Invigilation/Invigilator.py]: `datetime` is not needed here, revert.

File: algorithms/Invigilation/main.py [hunk 4]
```diff
@@ -297,6 +304,7 @@ def assign_invigilators(master_map, invigilator_list):
 
             try:
                 start, end = get_dates_from_key(time_slot_key)
+                #print("$$$",start, end)
```
Comment (Kailash0311) [inline algorithms/Invigilation/main.py]: This log statement needs to be removed

File: algorithms/Mailer/main.py [hunk 5]
```diff
@@ -37,14 +37,14 @@ def send_mails(subject, body, path):
                 continue
 
             email = get_email_from_path(_file)
+            # email = "f20160628@hyderabad.bits-pilani.ac.in"
```
Comment (Kailash0311) [inline algorithms/Mailer/main.py]: Changes in this file too need to be reverted.

File: algorithms/Mailer/mime_creator.py [hunk 6]
```diff
@@ -23,7 +23,7 @@ def create_message_with_attachment(to, subject, message_text, file):
         content_type = 'application/octet-stream'
 
     main_type, sub_type = content_type.split('/', 1)
-
+    print(main_type, sub_type)
```
Comment (Kailash0311) [inline algorithms/Mailer/mime_creator.py]: The log statements in this file need to be reverted.

File: final.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import csv
```
Comment (Kailash0311) [inline final.py]: This file is not needed.

**Comment by Kailash0311** [comment]
This PR does the following (according to the priority of issues it fixes): 
1. Allots secondary invigilator when a primary invigilator is not found. 
2. Changes the date format ( which is being imported from excel ) in most of the places to 
`"%d-%m-%y"` which is one of the standard date formats in the Excel. 
3. Puts try catch block in various places. This will ensure that the errors/discrepancies are easily traced.

**Pull request merged**